app/endpoints/questions.py

```diff
@@ -0,0 +1,45 @@
+from fastapi import APIRouter, HTTPException
+from pydantic import BaseModel
+import logging
+from app.services.database_manager import search_similar_texts
+from transformers import pipeline
+
+
+router = APIRouter()
+
+
+class QueryRequest(BaseModel):
+    question: str
+
+
+# Initialize the QA pipeline
+qa_pipeline = pipeline("question-answering")
+
+
+@router.post("/query")
```

Comment:
Also here, you can define the response schema:

{"question": question, "answer": answer["answer"]}

```python
Class EndpointResponse(BaseModel):
 question: str
 answer : str

```
This type of definitions with pydantic will create proper typing checks for all the outputs and avoid integration issues if something changes.